addons/stock_account/models/stock.py

```diff
@@ -435,7 +435,7 @@ def product_price_update_before_done(self, forced_qty=None):
         # adapt standard price on incomming moves if the product cost_method is 'average'
         std_price_update = {}
         for move in self.filtered(lambda move: move._is_in() and move.product_id.cost_method == 'average'):
-            product_tot_qty_available = move.product_id.qty_available + tmpl_dict[move.product_id.id]
+            product_tot_qty_available = move.product_id.with_context(owner_id=False).qty_available + tmpl_dict[move.product_id.id]
             rounding = move.product_id.uom_id.rounding
 
             qty_done = move.product_uom._compute_quantity(move.quantity_done, move.product_id.uom_id)
```

Comment:
I guess we have the same issue here. In a move with 2 move lines (1 with owner 1 without) the quantity_done should not be taken entirely